[CORE-6] Build error on Windows

When building Realm JS for node/Windows I get the following error:

```
c:\jenkins\workspace\realm_realm-js_pr-2495@4\src\object-store\src\sync\sync_session.cpp(422): error C2440: '=': cannot convert from 'std::_Array_iterator<_Ty,64>' to 'const char *' 
```

--- a/src/sync/sync_session.cpp
+++ b/src/sync/sync_session.cpp
@@ -419,7 +419,7 @@ SyncSession::SyncSession(SyncClient& client, std::string realm_path, SyncConfig 
         DBOptions options;
 
         if (m_config.realm_encryption_key) {
-            options.encryption_key = m_config.realm_encryption_key->begin();
+            options.encryption_key = m_config.realm_encryption_key->data();
         }
 #if REALM_ENABLE_SYNC
         auto history = sync::make_client_replication(m_realm_path);

